src/player.ts

```diff
@@ -2470,6 +2470,26 @@ export default class Player extends FakeEventTarget {
    */
   public _updateTracks(tracks: Array<Track>): void {
     Player._logger.debug('Tracks changed', tracks);
+
+    if (this.config.playback.updateAudioDescriptionFlavors && this._sources.metadata?.audioFlavors) {
+      let audioTracksIndex = 0;
+      // iterate over the audio tracks and set the isAudioDescription flag based on the audioFlavors metadata
+      tracks.forEach((track) => {
```

Comment:
I think the most generic solution, to prepare for a future case of 
"if we have a track with metadata A, then do logic B"
would be to extract the logic handling of handling the track to a separate file,
and then just leave the condition here

so something like
if (this.config.playback.updateAudioDescriptionFlavors) {
   audioDescriptionTrackHandler(track);
}

we could also make it even more generic and add a factory, but that's less important if we only have one usecase like this so far